Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -469,5 +492,55 @@ async Task<bool> IsLinuxGoogleBiosAsync()
                 return productName == "Google" || productName == "Google Compute Engine";
             }
         }
+
+        private async static Task<string> GetComputeEngineUniverseDomainUncachedAsync()
+        {
+            Logger.Info("Attempting to fetch the universe domain from the metadata server.");
+
+            // Using the built-in HttpClient, as we want bare bones functionality - we'll control retries.
```

Comment:
And no explicit proxy server support, of course... presumably that's okay, if we're running on Compute?

Reply:
Yes, this is the same we have for detecting Compute residency. I didn't want to make such changes here, but if we ever revisit, we need to look both here and con residency detection.